src/settings.c

```diff
@@ -463,6 +463,21 @@ struct golioth_settings *golioth_settings_init(struct golioth_client *client)
     return gsettings;
 }
 
+enum golioth_status golioth_settings_deinit(struct golioth_client *client,
```

Comment:
Same comment as the RPC PR: We can extract `client` from `settings` so this function only needs to take one argument.